### ISSUE TITLE
Visual inconsistency between demo and paper figure

### ISSUE THREAD

**Issue by xiao7199** [frank-xwang/UnSAM#10]
Hi, 

Thanks for open-sourcing your amazing work.  However, I notice an inconsistency between the image generated from demo_dico.ipynb 
![download](https://github.com/user-attachments/assets/3192f405-9559-4d1e-b6f3-631ef1f9c554)
and the pseudo label at Figure 3 of the paper
![img](https://github.com/user-attachments/assets/e1eff4e9-81a5-4a26-910e-23970cf14434)
Is there any additional step after demo_dico.ipynb to produce pseudo labels with the same quality as in Figure3?

Thanks!

**Comment by sslneurips2024** [comment]
Hello, thank you for your interest in our work. The demos in the figure illustrate the hierarchical structure of the masks predicted by our Divide-and-Conquer (DiCo) approach. In this visualization, the coarse-grained masks are shown in column 3 and the fine-grained masks in column 5. These are just a subset of what our model is capable of predicting. If you are looking to obtain "cleaner" results, you have the option to adjust the hyper-parameters of the DiCo pipeline. However, it's important to note that there is an inherent trade-off between recall and precision. Given that DiCo is primarily used to generate pseudo-masks for training segmentation models, we have observed that using pseudo-masks with higher recall tends to enhance the performance of the segmentation models trained on them. Therefore, we open-sourced the version and hyper-parameter settings with the best recall and precision trade-off for the second self-training stage. 

Hope this can answer your questions.

**Comment by frank-xwang** [comment]
Hello, thank you for your interest in our work. The demos in the figure illustrate the hierarchical structure of the masks predicted by our Divide-and-Conquer (DiCo) approach. In this visualization, the coarse-grained masks are shown in column 5 and the fine-grained masks in column 3. These are just a subset of masks that our model is capable of predicting. If you are looking to obtain "cleaner" results, you have the option to adjust the hyper-parameters of the DiCo pipeline. However, it's important to note that there is an inherent trade-off between recall and precision. Given that DiCo is primarily used to generate pseudo-masks for training segmentation models, we have observed that using pseudo-masks with higher recall tends to enhance the performance of the segmentation models trained on them. therefore, the open-sourced pipeline and settings focus on getting the best trade-off between recall and precision for the self-training stage.

**Comment by frank-xwang** [comment]
closing for now. please feel free to re-open this issue if you have more questions
